backend/elasticsearch/src/test/java/org/hibernate/search/backend/elasticsearch/orchestration/impl/ElasticsearchDefaultWorkSequenceBuilderTest.java

```diff
@@ -46,13 +48,18 @@ public void initMocks() {
 	@Test
 	@SuppressWarnings({ "rawtypes", "unchecked" })
 	public void simple() {
-		ElasticsearchWork<Void> work1 = work( 1 );
-		BulkableElasticsearchWork<Void> work2 = bulkableWork( 2 );
+		ElasticsearchWork<Object> work1 = work( 1 );
+		BulkableElasticsearchWork<Object> work2 = bulkableWork( 2 );
+
+		Object work1Result = new Object();
+		Object work2Result = new Object();
 
 		CompletableFuture<?> previousFuture = new CompletableFuture<>();
```

Comment:
We have lot of futures, maybe more then we really need. It is out of scope of the issue improving the design here. But I think that it would be nice whether we divided the declaration of the futures provided to the test subject, such as `flushFuture`, `work1Future`, `previousFuture`, from the ones that are created by the test subject itself, like `work1FutureFromSequenceBuilder` or `sequenceFuture`. What do you think of?

Reply:
Done